typescript/api/controllers/TranslationController.ts

```diff
@@ -0,0 +1,202 @@
+          };
+        }));
+
+        list.sort((a, b) => a.displayName.localeCompare(b.displayName));
+        return res.json(list);
+      } catch (err) {
+        sails.log.error('Error in TranslationController.getLanguages:', err);
+        return res.serverError(err);
+      }
+    }
+
+    /**
+     * Angular app endpoints (CSRF-enabled by default)
+     */
+    public async listEntriesApp(req, res) {
+      try {
+        const brandingName = req.params.branding;
+        const branding = BrandingService.getBrand(brandingName);
+        if (!branding) return res.badRequest({ message: `Unknown branding: ${brandingName}` });
+        const locale = req.param('locale');
+        const namespace = req.param('namespace') || 'translation';
+        const keyPrefix = req.param('keyPrefix');
+        const entries = await I18nEntriesService.listEntries(branding, locale, namespace, keyPrefix);
+        return res.json(entries);
+      } catch (err) {
+        sails.log.error('Error in TranslationController.listEntriesApp:', err);
+        return res.serverError(err);
+      }
+    }
+
+    public async setEntryApp(req, res) {
+      try {
+        const brandingName = req.params.branding;
+        const branding = BrandingService.getBrand(brandingName);
+        if (!branding) return res.badRequest({ message: `Unknown branding: ${brandingName}` });
+        const locale = req.param('locale');
+        const namespace = req.param('namespace') || 'translation';
+        const key = req.param('key');
+        const value = req.body?.value;
+        const category = req.body?.category;
+        const description = req.body?.description;
+        const saved = await I18nEntriesService.setEntry(branding, locale, namespace, key, value, { category, description });
+        try { TranslationService.reloadResources(); } catch (e) { sails.log.warn('[TranslationController.setEntryApp] reload failed', e?.message || e); }
+        return res.json(saved);
+      } catch (err) {
+        sails.log.error('Error in TranslationController.setEntryApp:', err);
+        return res.serverError(err);
+      }
+    }
+
+    public async getBundleApp(req, res) {
+      try {
+        const brandingName = req.params.branding;
+        const branding = BrandingService.getBrand(brandingName);
+        if (!branding) return res.badRequest({ message: `Unknown branding: ${brandingName}` });
+        const locale = req.param('locale');
+        const namespace = req.param('namespace') || 'translation';
+        const bundle = await I18nEntriesService.getBundle(branding, locale, namespace);
+        if (!bundle) return res.notFound({ message: 'Bundle not found' });
+        return res.json(bundle);
+      } catch (err) {
+        sails.log.error('Error in TranslationController.getBundleApp:', err);
+        return res.serverError(err);
+      }
+    }
+
+    public async setBundleApp(req, res) {
+      try {
+        const brandingName = req.params.branding;
```

Comment:
Something for later: there's a lot of duplicated code for getting the branding and other request params. They could be collapsed into a private helper method.